### ISSUE TITLE
07-naive-bayes-playstore-reviews.ipynb

### ISSUE THREAD

**Issue by mezcolantriz** [4GeeksAcademy/betaniammc-machine-learning#5]
Hola @Betaniammc , por aquí las issues!

## LO QUE HAS HECHO MUY BIEN

- **Preprocesamiento correcto**: `.str.strip().str.lower()` aplicado correctamente.

- **Probaste los 3 modelos NB**: MultinomialNB, BernoulliNB y GaussianNB. ¡Completo!

- **Análisis profundo de overfitting**: Excelente que te preocupaste por la diferencia train/test y no solo por el accuracy más alto.

- **Búsqueda manual de alpha**: Tu enfoque iterativo con el bucle `for alpha in np.arange(...)` es muy interesante y demuestra pensamiento analítico.

- **Métrica de "score" personalizada**: `df_res["score"] = df_res["acc_test"] - df_res["diff"]` - Muy creativo para balancear accuracy vs overfitting.

- **PASO 6 MUY COMPLETO**: Probaste Random Forest, AdaBoost, XGBoost, LightGBM y GradientBoosting. Además optimizaste TODOS con GridSearchCV. Trabajo muy extenso.

- **Observaciones constantes**: Después de cada modelo explicas qué observas. 

- **Conclusión final argumentada**: Comparaste todos los modelos y justificaste tu elección.

---

### **1. PASO 5 NO COMPLETADO - Guardar el modelo**
El enunciado pedía:
> *"Almacena el modelo en la carpeta correspondiente."*

No guardaste ningún modelo con pickle, a pesar de haberlo importado.

---

### **2. Imports excesivos e innecesarios** Una buena práctica es dejar solo los que uses :) 

Tienes imports que no usas para este ejercicio:
- `geopandas`, `contextily` (para mapas)
- `KMeans`, `KElbowVisualizer`, `silhouette_score` (para clustering)
- `make_blobs`

Esto hace el código más pesado y confuso. Importa solo lo que necesitas.

---

### **3. Clasificación binaria no multiclase**
```python
xgb = XGBClassifier(..., eval_metric='mlogloss')
```

`mlogloss` es para multiclase. Para binario mejor usar `logloss` o `binary:logistic`.

---

**Excelente trabajo, Betania.** Este es uno de los ejercicios más completos que he revisado. Tu análisis de overfitting y la búsqueda manual de alpha demuestran pensamiento crítico. ¡Muy bien!
